src/redux/books/books.js

```diff
@@ -1,34 +1,51 @@
-import { v4 } from 'uuid';
+export const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/cztRrb2hdveZJBbq0L1F/';
 
 const ADD = 'bookstore/books/ADD';
 const REMOVE = 'bookstore/books/REMOVE';
+const FETCH_BOOKLIST = 'bookstore/books/FETCH_BOOKLIST';
 
-const defaultBooks = [
-  { id: v4(), author: 'George Martin', title: 'Game Of Thrones' },
-  { id: v4(), author: 'Mario Mendoza', title: 'Satanas' },
-];
-
-export default function reducer(state = defaultBooks, action) {
+export default function reducer(state = [], action) {
+  const bookList = [];
   switch (action.type) {
     case ADD:
       return [...state, action.book];
     case REMOVE:
-      return state.filter((book) => book.id !== action.id);
+      return state.filter((book) => book.item_id !== action.item_id);
+    case FETCH_BOOKLIST:
+      Object.values(action.data).forEach((item) => {
+        bookList.push(item[0]);
+      });
+      return bookList;
     default:
       return state;
   }
 }
 
 export function addBook(book) {
-  return {
-    type: ADD,
-    book,
+  return async (dispatch) => {
+    await fetch(`${url}books`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+      body: JSON.stringify(book),
+    }).then(() => dispatch({ type: ADD, book }));
   };
 }
 
 export function removeBook(id) {
-  return {
-    type: REMOVE,
-    id,
+  return async (dispatch) => {
+    await fetch(`${url}books/${id}`, {
+      method: 'DELETE',
+    }).then(() => dispatch({ type: REMOVE, item_id: id }));
+  };
+}
+
+export function getBookList() {
```

Comment:
- [ ] It would be better if your app fetch all the books that are in the API. Currently, your `getBookList` function does not retrieve the books from the API. kindly fix this minor issue. 

![Capture](https://user-images.githubusercontent.com/64173189/162278673-70a16ed0-ac83-4ee4-89d4-97aab1332922.JPG)


![Capture](https://user-images.githubusercontent.com/64173189/162278607-29c92038-65b3-4053-a257-f25ce57f106a.JPG)

Reply:
Hello @BrianSammit,  regarding this issue please read the comment in the pull request. 
You see, the function is working to retrieve the data from the API, but when it gets to the `dispatch()` part, it says that it is not a function. Would you please provide me feedback about this?

Reply:
- If that is the case I will advice that all [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) calls are done by you trying to use [axios](https://axios-http.com/docs/intro). What do you think?